the_bot_engine/src/main/java/org/freakz/services/foreca/ForecaWeatherService.java

```diff
@@ -314,6 +306,15 @@ public List<CountryCityLink> getMatchingCountryCityLinks(String place) {
             if (countryCityLink.city.toLowerCase().matches(place) || countryCityLink.city2.toLowerCase().matches(place)) {
                 matching.add(countryCityLink);
             }
+            else{
+                String[] pieces = place.split("/");
```

Comment:
This can be moved before for loop so it is done only once and not everytime in loop.
Also do toLowerCase() before split then you can make matching compare cleaner and remove toLowerCase() there.